ui/apps/platform/src/Containers/Collections/RuleSelector/ByLabelSelector.tsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import {
+    Flex,
+    Label,
+    FormGroup,
+    FlexItem,
+    Button,
+    Divider,
+    ValidatedOptions,
+} from '@patternfly/react-core';
+import { TrashIcon } from '@patternfly/react-icons';
+import { FormikErrors } from 'formik';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { SelectorEntityType, ScopedResourceSelector, ByLabelResourceSelector } from '../types';
+import { AutoCompleteSelect } from './AutoCompleteSelect';
+
+export type ByLabelSelectorProps = {
+    entityType: SelectorEntityType;
+    scopedResourceSelector: ByLabelResourceSelector;
+    handleChange: (
+        entityType: SelectorEntityType,
+        scopedResourceSelector: ScopedResourceSelector
+    ) => void;
+    validationErrors: FormikErrors<ByLabelResourceSelector> | undefined;
+};
+
+function ByLabelSelector({
+    entityType,
+    scopedResourceSelector,
+    handleChange,
+    validationErrors,
+}: ByLabelSelectorProps) {
+    function onChangeLabelKey(resourceSelector: ByLabelResourceSelector, ruleIndex) {
```

Comment:
A possible follow up is simple naming convention to make it even clearer to future readers that these are higher-order functions which return callback functions.

Reply:
I think this is a leftover from an earlier refactor. I'll see if I can get rid of the HOF in favor of a plain callback, like the other event handlers.